javaslang/src/test/java/javaslang/collection/euler/Euler43Test.java

```diff
@@ -54,25 +54,22 @@ public void shouldSolveProblem43() {
     private static Seq<Long> tenDigitPandigitalsWithProperty() {
         final CharSeq ALL_DIGITS = CharSeq.of("0123456789");
         final List<Integer> DIVISORS = List.of(2, 3, 5, 7, 11, 13, 17);
-        final char DUMMY_CHAR = Character.MIN_VALUE;
 
         return ALL_DIGITS
                 .combinations(2)
                 .flatMap(CharSeq::permutations)
-                .map(cs -> Tuple.of(cs, cs.prepend(DUMMY_CHAR), Integer.valueOf(cs.mkString())))
                 .flatMap(firstTwoDigits -> DIVISORS
                         .foldLeft(List.of(firstTwoDigits), (accumulator, divisor) -> accumulator
                                 .flatMap(digitsSoFar -> ALL_DIGITS
-                                        .removeAll(digitsSoFar._1)
-                                        .map(nextDigit -> Tuple.of(digitsSoFar._1.append(nextDigit), digitsSoFar._2.tail().append(nextDigit)))
-                                        .map(digitsToTest -> Tuple.of(digitsToTest._1, digitsToTest._2, Integer.valueOf(digitsToTest._2.mkString())))
+                                        .removeAll(digitsSoFar)
+                                        .map(nextDigit -> digitsSoFar.append(nextDigit))
                                 )
-                                .filter(digitsToTest -> digitsToTest._3 % divisor == 0)
+                                .filter(digitsToTest -> Integer.valueOf(digitsToTest.takeRight(3).mkString()) % divisor == 0)
```

Comment:
To avoid boxing/unboxing

```java
Integer.valueOf(digitsToTest.takeRight(3).mkString()) % divisor == 0
```

is converted best using

```java
digitsToTest.takeRight(3).parseInt() % divisor == 0
```